paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetLoader.kt

```diff
@@ -58,42 +57,43 @@ internal class DefaultCustomerSheetLoader(
         customerAdapterProvider = CustomerSheetHacks.adapter,
     )
 
-    override suspend fun load(configuration: CustomerSheet.Configuration?): Result<CustomerSheetState.Full> {
-        val customerAdapter = customerAdapterProvider.awaitAsResult(
-            timeout = 5.seconds,
-            error = {
-                "Couldn't find an instance of CustomerAdapter. " +
-                    "Are you instantiating CustomerSheet unconditionally in your app?"
-            },
-        )
+    override suspend fun load(configuration: CustomerSheet.Configuration): Result<CustomerSheetState.Full> {
+        return runCatching {
+            val customerAdapter = customerAdapterProvider.awaitAsResult(
+                timeout = 5.seconds,
+                error = {
+                    "Couldn't find an instance of CustomerAdapter. " +
+                        "Are you instantiating CustomerSheet unconditionally in your app?"
+                },
+            ).getOrThrow()
 
-        return customerAdapter.mapCatching { adapter ->
-            if (adapter.canCreateSetupIntents) {
-                adapter to retrieveElementsSession(configuration, adapter).getOrThrow()
-            } else {
-                adapter to null
-            }
-        }.map { (adapter, elementsSessionWithMetadata) ->
+            val elementsSessionWithMetadata = retrieveElementsSession(configuration, customerAdapter).getOrThrow()
             loadPaymentMethods(
-                customerAdapter = adapter,
+                customerAdapter = customerAdapter,
                 configuration = configuration,
                 elementsSessionWithMetadata = elementsSessionWithMetadata,
             ).getOrThrow()
         }
     }
 
     private suspend fun retrieveElementsSession(
-        configuration: CustomerSheet.Configuration?,
+        configuration: CustomerSheet.Configuration,
         customerAdapter: CustomerAdapter,
     ): Result<ElementsSessionWithMetadata> {
+        val paymentMethodTypes = if (customerAdapter.canCreateSetupIntents) {
+            customerAdapter.paymentMethodTypes ?: emptyList()
```

Comment:
what's the user experience if we return empty list here? and why would paymentMethodTypes == null?

Reply:
`IntentConfiguration` (where this is used below) only accepts a non null param. But I think in practice the behavior would be the same.

empty list/null is our default implementation for this, and just results in no filtering done by v1/elements/sessions.

Reply:
got it